rerun_cpp/src/rerun/component_batch.hpp

```diff
@@ -156,7 +155,19 @@ namespace rerun {
 
         /// Move assignment
         void operator=(ComponentBatch<TComponent>&& other) {
+// Need to disable the maybe-uninitialized here.  It seems like the compiler may be confused in situations where
+// we are assigning into an unused optional from a temporary. The fact that this hits the move-assignment without
+// having called the move constructor is suspicious though and hints of an actual bug.
+//
+// See: https://github.com/rerun-io/rerun/issues/4027
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
```

Comment:
or maybe make reusable macros for this, i.e.

```C
#define RERUN_PUSH_DIAGNOSTICS_OFF                            \
    #ifdef __GNUC__                                           \
    #  pragma GCC diagnostic push                             \
    #  pragma GCC diagnostic ignored "-Wmaybe-uninitialized"  \
    #endif

#define RERUN_POP_DIAGNOSTICS_TWEAK \
    #ifdef __GNUC__                 \
    #  pragma GCC diagnostic pop  \
    #endif
#endif
```

that we can then call where we need it. This is just so noisy